match/match.go

```diff
@@ -73,6 +78,35 @@ func matchCreateHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	userOne := *req.UserOne
+	userTwo := *req.UserTwo
+
+	userOneValid, err := coms.CheckUser(userOne)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Unable to reach %s service: %s", "matches", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	userTwoValid, err := coms.CheckUser(userTwo)
+	if err != nil {
+		errMsg := utils.CreateErrorJSON(fmt.Sprintf("Unable to reach %s service: %s", "matches", err.Error()))
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	if !userOneValid {
```

Comment:
I suggest moving this check to straight after the `checkUser` call for `userOne`, so the second `checkUser` call for `userTwo` isn't made if `userOne` isn't valid.